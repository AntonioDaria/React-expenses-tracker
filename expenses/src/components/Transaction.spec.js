import React from 'react';
import { render } from '@testing-library/react';
import Transaction from './Transaction.js'

const MOCK_TRANSACTIONS = 
    [
        { id: 1, text: 'Flower', amount: -20 },
        { id: 2, text: 'Salary', amount: 300 },
        { id: 3, text: 'Book', amount: -10 },
        { id: 4, text: 'Camera', amount: 150 }
    ]


describe('Transaction component', () => {
    let container;


    beforeEach(() => {
        ({ container} = renderTransaction());
    });

    it('should match the snapshot', () => {
        expect(container).toMatchSnapshot();
    });
});

const renderTransaction = (mockTransactions = MOCK_TRANSACTIONS) => {
    return render(
        <Transaction transaction={mockTransactions} />
    );
};
