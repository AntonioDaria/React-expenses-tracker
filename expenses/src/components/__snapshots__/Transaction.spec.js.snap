// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component should match the snapshot 1`] = `
<div>
  <li
    class="minus"
  >
     
    <span>
      +
      $
      NaN
    </span>
    <button
      class="delete-btn"
    >
      x
    </button>
  </li>
</div>
`;
